### ISSUE TITLE
Missing transformer config for PodTemplate envFrom

### ISSUE THREAD

**Issue by vdboor** [kubernetes-sigs/kustomize#4554]
The following transformer config seems to be missing for built-in Kubernetes resources:

```
nameReference:
  - kind: ConfigMap
    fieldSpecs:
      - kind: PodTemplate
        path: template/spec/containers/envFrom/configMapRef
```

Currently, when deploying a `PodTemplate` resource, the `configMapRef` field is not adjusted for any resource name changes. Looking at https://github.com/kubernetes-sigs/kustomize/tree/master/examples/transformerconfigs#builtin-namereference more entries might be missing.

FYI, I've been using the `PodTemplate` resource to run additional tasks from manual CI pipelines. The PodTemplate serves as the pod boilerplace so `kubectl run` can create a pod with the proper configuration. While doing so, it appeared that the `envFrom.configMapRef:` was not properly replaced.

**Comment by CandiedCode** [comment]
looks like there was a PR fix your issue recently.

https://github.com/kubernetes-sigs/kustomize/commit/e97cecf9dce84642b60b47c44df1c8d39425ac79


I think this should solve your issue and we can close this issue.

**Comment by KnVerey** [comment]
This should indeed be fixed in kustomize v4.5.4. Please reopen if that is not the case.

/close
